packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -8,27 +8,34 @@ import { useRef } from '@wordpress/element';
  */
 import useRefEffect from '../use-ref-effect';
 
-/** @typedef {import('@wordpress/element').RefCallback} RefCallback */
-
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @template T
+ * @param {T} value
+ * @return {import('react').MutableRefObject<T>} A ref with the value.
+ */
 function useFreshRef( value ) {
-	const ref = useRef();
-	ref.current = value;
+	/* eslint-enable jsdoc/valid-types */
+	/* eslint-disable jsdoc/no-undefined-types */
+	/** @type {import('react').MutableRefObject<T>} */
+	/* eslint-enable jsdoc/no-undefined-types */
+	const ref = useRef( value );
 	return ref;
 }
 
 /**
  * A hook to facilitate drag and drop handling.
  *
- * @param {Object}     $1             Named parameters.
- * @param {boolean}    $1.isDisabled  Whether or not to disable the drop zone.
- * @param {DragEvent}  $1.onDragStart Called when dragging has started.
- * @param {DragEvent}  $1.onDragEnter Called when the zone is entered.
- * @param {DragEvent}  $1.onDragOver  Called when the zone is moved within.
- * @param {DragEvent}  $1.onDragLeave Called when the zone is left.
- * @param {MouseEvent} $1.onDragEnd   Called when dragging has ended.
- * @param {DragEvent}  $1.onDrop      Called when dropping in the zone.
+ * @param {Object} props Named parameters.
+ * @param {boolean} props.isDisabled Whether or not to disable the drop zone.
+ * @param {(e: DragEvent) => void} props.onDragStart Called when dragging has started.
+ * @param {(e: DragEvent) => void} props.onDragEnter Called when the zone is entered.
+ * @param {(e: DragEvent) => void} props.onDragOver Called when the zone is moved within.
+ * @param {(e: DragEvent) => void} props.onDragLeave Called when the zone is left.
+ * @param {(e: MouseEvent) => void} props.onDragEnd Called when dragging has ended.
```

Comment:
Should this be `DragEvent` too?

Reply:
`onDragEnd` comes from `onmouseup` so I think it should be a `MouseEvent`.